src/Servers/Kestrel/Core/src/Internal/Http/Http1OutputProducer.cs

```diff
@@ -118,31 +116,27 @@ public ValueTask<FlushResult> WriteDataToPipeAsync(ReadOnlySpan<byte> buffer, Ca
 
         public ValueTask<FlushResult> WriteStreamSuffixAsync()
         {
+            ValueTask<FlushResult> result = default;
+
             lock (_contextLock)
             {
-                if (_writeStreamSuffixCalled)
+                if (!_writeStreamSuffixCalled)
                 {
-                    // If WriteStreamSuffixAsync has already been called, no-op and return the previously returned ValueTask.
-                    return _writeStreamSuffixValueTask;
-                }
+                    if (_autoChunk)
+                    {
+                        var writer = new BufferWriter<PipeWriter>(_pipeWriter);
+                        result = WriteAsyncInternal(ref writer, EndChunkedResponseBytes);
+                    }
+                    else if (_unflushedBytes > 0)
+                    {
+                        result = FlushAsync();
+                    }
 
-                if (_autoChunk)
-                {
-                    var writer = new BufferWriter<PipeWriter>(_pipeWriter);
-                    _writeStreamSuffixValueTask = WriteAsyncInternal(ref writer, EndChunkedResponseBytes);
-                }
-                else if (_unflushedBytes > 0)
-                {
-                    _writeStreamSuffixValueTask = FlushAsync();
-                }
-                else
-                {
-                    _writeStreamSuffixValueTask = default;
+                    _writeStreamSuffixCalled = true;
```

Comment:
@halter73, this is the change I pinged you about.  Please let me know if it looks ok.

Reply:
LGTM. Thanks for doing this!

Reply:
> LGTM. Thanks for doing this!

You're welcome.  My motivation is I expect the moment I merge https://github.com/dotnet/runtime/pull/50116, we're going to start seeing that attribute get slapped into a bunch of places in this codebase :)